hbase-server/src/main/java/org/apache/hadoop/hbase/util/FSTableDescriptors.java

```diff
@@ -123,8 +136,10 @@ public static TableDescriptor tryUpdateAndGetMetaTableDescriptor(Configuration c
     FileSystem fs, Path rootdir) throws IOException {
     // see if we already have meta descriptor on fs. Write one if not.
     try {
-      return getTableDescriptorFromFs(fs, rootdir, TableName.META_TABLE_NAME);
-    } catch (TableInfoMissingException e) {
+      TableDescriptor td = getTableDescriptorFromFs(fs, rootdir, TableName.META_TABLE_NAME);
+      validateMetaTableDescriptor(td);
+      return td;
+    } catch (TableInfoMissingException  | NoSuchColumnFamilyException e) {
```

Comment:
This makes sense. The table might be missing altogether (what current code handles), or might be missing a column family due to legacy (what this change handles)